assembly/assemblyCommon/AssemblyAssemblyV2.cc

```diff
@@ -1344,9 +1351,20 @@ void AssemblyAssemblyV2::ReturnToPlatform_start()
     return;
   }
 
+  if(!position_z_before_makespace_stored_)
+  {
+    NQLog("AssemblyAssemblyV2", NQLog::Warning) << "ReturnToPlatform_start"
+       << ": stage position before making space on platform has not been stored. Cannot continue.";
+    QMessageBox::warning(0, tr("[LStepExpressMotionManager]"),
+       QString("Stage position before making space on platform has not been stored. Cannot continue."),
+       QMessageBox::Abort
+    );
+    return;
+  }
+
   const double dx0 = 0.0;
   const double dy0 = 0.0;
-  const double dz0 = -makespace_Z;
+  const double dz0 = position_z_before_makespace_ - motion_->get_position_Z();
```

Comment:
I guess you tested that the sign is correct. It's hard to say.

Reply:
Yes, I tested this :D